File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        // solveing using kadans algorithm
        int n=nums.size();
        int currMax=nums[0];
        int maxSum=nums[0];
        for(int i=1;i<n;i++){
            currMax=max(nums[i],currMax+nums[i]);
            maxSum=max(currMax,maxSum);
        }
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
import java.util.*;
class Solution {
    public int maxSubArray(int[] nums) {
        int n=nums.length;
        int maxi=-1000000000;
        int currMax=-1000000000;
        for(int i=0;i<n;i++){
            currMax=Math.max(nums[i],currMax+nums[i]);
            maxi=Math.max(currMax,maxi);
        }
        return maxi;
    }
}